lib/ruby2ruby.rb

```diff
@@ -481,18 +490,27 @@ def process_hash(exp) # :nodoc:
     result = []
 
     until exp.empty?
-      lhs = process(exp.shift)
+      lhs = exp.shift
       rhs = exp.shift
-      t = rhs.first
-      rhs = process rhs
-      rhs = "(#{rhs})" unless [:lit, :str].include? t # TODO: verify better!
-
-      result << "#{lhs} => #{rhs}"
+      result << process_hash_pair(lhs, rhs)
     end
 
     return result.empty? ? "{}" : "{ #{result.join(', ')} }"
   end
 
+  def process_hash_pair(lhs, rhs)
```

Comment:
I agree with the name being bad, tho Id on't have a good suggestion. I don' think you gain much with this refactoring. I'd be tempted to unfactor it back.